tests/resource/valid/collection/dictionary_return_is_optional_check.py

```diff
@@ -0,0 +1,5 @@
+from typing import Optional
+j = { 10: 200, 30: 5 }
+
+def f(key: int) -> Optional[int]:
+    return j[key] if key in j else None
```

Comment:
Actually I don't think we should modify the behaviour here.
Instead, we should deal with `KeyError`.
For now, this error is unchecked then in Mamba.
However, as soon as we have pseudo Python syntax (#392 ) we will be able to specify that for instance a dictionary access can give a `KeyError`, and we will be forced to deal with that.
I think it is too cumbersome to start clumsily implementing optionals.